backend/src/rate-limiter.js

```diff
@@ -1,76 +1,98 @@
 const moment = require('moment');
 const redisClient = require('./redis-client');
 
-const RATE_LIMIT_WINDOW_HOURS = 24; // Window duration for which rate limiting should apply
-const REQUEST_LIMIT_COUNT = 20000; // Max number of requests per window
+const RATE_LIMIT_WINDOW_HOURS = 3; // Window duration for which rate limiting should apply
+const REQUEST_LIMIT_COUNT = 250; // Max number of requests per window
 const LOG_GROUP_WINDOW_HOURS = 1; // Rate limit window bucket duration
 
-// Sliding window counter based rate limiting middleware
-const rateLimiterMiddleware = (req, res, next) => {
-  try {
-    if (!redisClient) {
-      throw new Error('Redis client not initialized');
+const shouldRateLimit = (rateLimitKey) => {
+  if (!redisClient) {
+    throw new Error('Redis client not initialized');
+  }
+  return redisClient.get(rateLimitKey, (err, val) => {
```

Comment:
should we be using user email and IP as keys in the same redis instance? Is there a way to separate the storage of different types of keys

Reply:
I don't see any issues with using the same redis instance. Emails and IP addresses don't conflict with each other.